dvc/commands/data.py

```diff
@@ -0,0 +1,364 @@
+                new = None
+
+            cache = self.repo.odb.local
+            root = str(out)
+            old = out.get_obj()
+            d = diff(root, old, new, cache, granular=granular)
+            for state, items in d.items():
+                if not items:
+                    continue
+                unstaged_diff[state].extend(items)
+
+        not_in_cache: List[str] = unstaged_diff.pop("not_in_cache", [])
+        unstage_unchanged: Set[str] = set(unstaged_diff.pop("unchanged", []))
+
+        # we need to store objects from index and the HEAD to diff later
+        objs: Dict[str, Dict[str, "HashFile"]] = defaultdict(dict)
+        staged_diff: Dict[str, List[str]] = defaultdict(list)
+        try:
+            for _rev in self.repo.brancher(revs=[head]):
+                for out in self.repo.index.outs:
+                    out = cast("Output", out)
+                    if not out.use_cache:
+                        continue
+
+                    root = str(out)
+                    typ = "index" if _rev == "workspace" else head
+                    objs[root][typ] = out.get_obj()
+        except (SCMError, NoSCMError):
+            unchanged: Set[str] = unstage_unchanged
+        else:
+            for root, obj_d in objs.items():
+                old = obj_d.get(head, None)
+                new = obj_d.get("index", None)
+                cache = self.repo.odb.local
+                d = diff(root, old, new, cache, granular=granular)
+                for state, items in d.items():
+                    if not items:
+                        continue
+                    staged_diff[state].extend(items)
+
+            staged_diff.pop("not_in_cache", None)
+            unchanged = (
+                set(staged_diff.pop("unchanged", [])) & unstage_unchanged
+            )
+
+        git_info = self._git_info(untracked_files=untracked_files)
+        return {
+            "staged": staged_diff,
+            "unstaged": unstaged_diff,
+            "not_in_cache": not_in_cache,
+            "unchanged": unchanged,
+            "git": git_info,
+        }
+
+
+def add_parser(subparsers, parent_parser):
+    data_parser = subparsers.add_parser(
+        "data",
+        parents=[parent_parser],
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    data_subparsers = data_parser.add_subparsers(
+        dest="cmd",
+        help="Use `dvc data CMD --help` to display command-specific help.",
+    )
+    fix_subparsers(data_subparsers)
+    data_status_parser = data_subparsers.add_parser(
+        "status",
+        parents=[parent_parser],
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    data_status_parser.add_argument(
+        "--json", action="store_true", default=False
+    )
+    data_status_parser.add_argument(
+        "--granular",
+        action="store_true",
+        default=False,
+    )
+    data_status_parser.add_argument(
+        "--unmodified",
```

Comment:
I find `unmodified` to be a bit strange, how about using `unchanged` instead?

Reply:
`unchanged` makes sense to me as well, as in `change -> replace` and `modify -> update`. We have `modified` to mean that something was updated, but `unmodified` is not quite the opposite of `modified`, it's the opposite of `modified + added + deleted + ...` etc, as in not replaced/changed.

(Note that it's how it works internally, I have been remapping on the CLI.) 

cc @dberenbaum

Reply:
No strong opinion on the name here. I'm fine with `unchanged` if you both prefer.